### PR TITLE
Resolve unblocked TODOs and formatting changes to System.Net.Mail.

### PR DESCRIPTION
**Pull request by Priya91**
Address PR comments left by @CIPop 

Also addresses TODO with issue #11747

cc @CIPop @stephentoub

### CODE REVIEW

File: src/Common/src/System/Net/Logging/EmailEventSource.cs [hunk 1]
```diff
@@ -10,18 +10,18 @@
 
 namespace System.Net
 {
-    [EventSource(Name = "Microsoft-System-Net-Mail", LocalizationResources = "FxResources.System.Net.Mail.SR")]
-    internal sealed class WebEventSource : EventSource
+    [EventSource(Name = "Microsoft-System-Net-Mail", Guid = "32663b9f-4d8d-44f8-9f29-f4efc12e6879", LocalizationResources = "FxResources.System.Net.Mail.SR")]
+    internal sealed class EmailEventSource : EventSource
```
Comment (stephentoub) [inline src/Common/src/System/Net/Logging/EmailEventSource.cs]: EmailEventSource => MailEventSource?

File: src/Common/src/System/Net/Logging/EmailEventSource.cs [hunk 2]
```diff
@@ -10,18 +10,18 @@
 
 namespace System.Net
 {
-    [EventSource(Name = "Microsoft-System-Net-Mail", LocalizationResources = "FxResources.System.Net.Mail.SR")]
-    internal sealed class WebEventSource : EventSource
+    [EventSource(Name = "Microsoft-System-Net-Mail", Guid = "32663b9f-4d8d-44f8-9f29-f4efc12e6879", LocalizationResources = "FxResources.System.Net.Mail.SR")]
```
Comment (stephentoub) [inline src/Common/src/System/Net/Logging/EmailEventSource.cs]: In the past I believe @vancem's guidance has been to not explicitly specify a Guid, and to not worry about localizing resources.

Comment (davidsh) [inline src/Common/src/System/Net/Logging/EmailEventSource.cs]: Even we chose not to localize the resources, we place all strings in the resource file anyways.  This is better than putting hard-coded strings in the *.CS files themselves.

So, I would prefer we keep this design as-is and leave strings in the resources file.

File: src/Common/src/System/Net/Logging/EmailEventSource.cs (hunk 1 above)
Comment (Priya91) [inline src/Common/src/System/Net/Logging/EmailEventSource.cs]: Oki, will rename.

File: src/Common/src/System/Net/Logging/EmailEventSource.cs (hunk 2 above)
Comment (davidsh) [inline src/Common/src/System/Net/Logging/EmailEventSource.cs]: >However you can also use *Microsoft-System-Net-Mail in most tools

What about tools where you have to specify the GUID?

And we already have GUIDs for other event sources.  See:
Documentation/debugging/windows-instructions.md

Comment (Priya91) [inline src/Common/src/System/Net/Logging/EmailEventSource.cs]: Will remove the GUID for now, @CIPop please file issue if you disagree. Thanks!

Comment (CIPop) [inline src/Common/src/System/Net/Logging/EmailEventSource.cs]: @Priya91 @vancem 

It's fine to remove the GUID but add it within both the source code as well as https://github.com/dotnet/corefx/blob/master/Documentation/debugging/windows-instructions.md#traces

> However you can also use *Microsoft-System-Net-Mail in most tools instead (which is MUCH nicer).

I found this doesn't work `logman`. 
@vancem please review https://github.com/dotnet/corefx/blob/master/Documentation/debugging/windows-instructions.md#traces and let me know if there is a way to use the `*<name>` format using existing Windows tools.

Comment (CIPop) [inline src/Common/src/System/Net/Logging/EmailEventSource.cs]: Tracking with #12808.

**Pull request merged**